app/views/devise/registrations/_registration_form.html.erb

```diff
@@ -14,12 +14,17 @@
     </div>
 
     <div class="registration__actions">
+      <% if params[:state] == "new-user" && !Settings::Authentication.allow_email_password_registration && !display_social_login? %>
```

Comment:
I almost feel this could be a helper with the explanation as a comment:

```ruby
# Display the fallback message if we can't register with email and at the same time can't display the social options.
def display_registration_fallback?(state)
  state == "new-user" && !Settings::Authentication.allow_email_password_registration && !display_social_login?
end

display_registration_fallback?(params[:state])
```